### ISSUE TITLE
ignoring unsupported keyword arguments for client and ExtraArgs

### ISSUE THREAD

**Issue by patrickleclair-GORDONFN** [RaRe-Technologies/smart_open#600]
#### Problem description

Be sure your description clearly answers the following questions:

- What are you trying to achieve?
   Using the `smart_open.open` function I am trying to pass the `client` and `ExtraArgs` keywords allowed in Boto3 to my stream to an S3 bucket. [boto3 upload docs](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/s3-uploading-files.html)
- What is the expected result?
  My expectation is that the data will be added to the s3 bucket without any warnings and that the ExtraArgs will be applied
- What are you seeing instead?
   What I am seeing instead is that while the data is still being uploaded to the s3 bucket it:
        1. Kicks back a warnings re: `client` and `ExtraArgs` keywords
        2. Doesn't apply the ExtraArgs to the dataset on s3
#### Steps/code to reproduce the problem

In order for us to be able to solve your problem, we have to be able to reproduce it on our end.
Without reproducing the problem, it is unlikely that we'll be able to help you.

**input**
```python

s3_client = boto3.client("s3")
extra_args = {'ContentType': 'text/csv'}
file_path = smart_open.open(("s3://" + bucket + "/" + key), 'w', newline="",
                                        transport_params=dict(client=s3_client, ExtraArgs=extra_args))

file = io.StringIO()
with file_path as file_out:
    writer = DictWriter(file)
    writer.writeheader()
    file_out.write(file.getvalue())
```

**output**
```python
WARNING - ignoring unsupported keyword arguments: ['ExtraArgs', 'client']
```

Include full tracebacks, logs and datasets if necessary.
Please keep the examples minimal ([minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example)).

#### Versions

Please provide the output of:

```python
import platform, sys, smart_open
print(platform.platform())
print("Python", sys.version)
print("smart_open", smart_open.__version__)
```

```
Windows-10-10.0.19041-SP0
Python 3.8.3 (tags/v3.8.3:6f8c832, May 13 2020, 22:37:02) [MSC v.1924 64 bit (AMD64)]
smart_open 4.2.0
```

#### Checklist

Before you create the issue, please make sure you have:

- [x] Described the problem clearly
- [x] Provided a minimal reproducible example, including any required data
- [x] Provided the version numbers of the relevant software

**Comment by mpenkov** [comment]
We don't support ExtraArgs, because we don't call the function that accepts them (upload_file). See https://github.com/RaRe-Technologies/smart_open/blob/develop/howto.md#how-to-pass-additional-parameters-to-boto3 for more details.

Does the warning go away if you remove ExtraArgs?

**Comment by patrickleclair-GORDONFN** [comment]
Thanks for the heads up regarding ExtraArgs. Is there thoughts on adding this as a feature eventually?

Also the warning for `client` still appears when ExtraArgs is removed.

**Comment by mpenkov** [comment]
No, because we use a different API to upload the file. What arguments are you attempting to pass? Is it possible to pass them as something compatible with the functions that we are calling instead?

Furthermore, the reason the client is being ignored is because you're incorrectly passing a tuple instead of a string as a URL.

Wrong:

```python
file_path = smart_open.open(("s3://" + bucket + "/" + key), 'w', newline="", ...)
```

Right:

```
file_path = smart_open.open("s3://" + bucket + "/" + key, 'w', newline="", ...)
```

**Comment by patrickleclair-GORDONFN** [comment]
For the Extra Args we are looking at setting the ServerSideEncryption and ContentType. I looked at the Boto3 docs and they have both of those as passable arguments for what appear to be all of the functions you call from the how-to docs you sent me. However when I put them in I get the same warning and they aren't added to my file in S3.

```python
s3_client = boto3.client("s3")
            file_path = smart_open.open("s3://" + bucket + "/" + key, 'w', newline="",
                                        transport_params=dict(client=s3_client, ContentType=extra_args['ContentType'],
                                                              ServerSideEncryption=extra_args['ServerSideEncryption']))
```
output:
```
smart_open.utils - WARNING - ignoring unsupported keyword arguments: ['ContentType', 'ServerSideEncryption', 'client']
```
I also tried running:

```python
s3_client = boto3.client("s3")
            file_path = smart_open.open("s3://" + bucket + "/" + key, 'w', newline="",
                                        transport_params=dict(client=s3_client))
```

and still got the error

```
smart_open.utils - WARNING - ignoring unsupported keyword arguments: ['client']
```

**Comment by mpenkov** [comment]
OK, I've worked out what the problem is. First and foremost, the docs were for an unreleased developer version, and you had 4.2.0 installed. I've since released 5.0.0 - if you install that, the behavior should be more like what you expected.

Second, you're passing the Content-Type parameter (and other parameters) incorrectly. You need to do something like this:

```python
transport_params = {
  'client_kwargs': {
     'S3.Client.put_object': {
       'Content-Encoding': ...,
     },
  },
}
```

See the [docs](https://github.com/RaRe-Technologies/smart_open/blob/develop/howto.md#how-to-pass-additional-parameters-to-boto3) for more examples.

**Comment by patrickleclair-GORDONFN** [comment]
Thanks for all of your help. I was able to pass the parameters and have it run without warnings now.
